contrib/container-chroot-agent/entrypoint-agent.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -x
+
+# mount FSs from the host VM to /chroot folder
+for i in proc sys run dev var; do mkdir -p /chroot/$i; done
+# special filesystem type: proc
+mount --types proc /rootproc /chroot/proc
+# for the following: sys, dev and var we only want changes to come from host to chroot, not the other way, so we use --make-slave
+mount --bind /rootsys /chroot/sys
+mount --make-slave /chroot/sys
+mount --bind /rootdev /chroot/dev
+mount --make-slave /chroot/dev
+mount --bind /rootrun /chroot/run
+mount --make-slave /chroot/run
+# so that our /var/cfengine doesn't show up on the host we bind mount as a slave
+mount --bind /rootfs/var /chroot/var
+mount --make-slave /chroot/var
+for i in bin lib lib64 sbin; do cp -a /rootfs/$i /chroot/; done
+
+# install cfengine in /cfengine in container and then bind mount to /chroot/var/cfengine so that contents are only visible/present in container/chroot and not host
+# hack to install cfengine-nova, logged ticket ENT-11403 to make quick-install script used later for installation
+# for now, contact sales if you are interested in acquiring needed tarballs
+if [ -f /dunfell-cfengine-nova.tar.gz ]; then
+  cd /
+  tar xf /dunfell-cfengine-nova.tar.gz
+  cd /cfengine
+  tar xf /liblmdb.tar.gz
+fi
+
+for i in $(ls -1 /rootfs/ | grep -v -E "bin|dev|lib|lib64|proc|run|sbin|sys|var"); do mkdir /chroot/$i && mount -o bind /rootfs/$i /chroot/$i; done
+
+mount -o bind /cfengine /chroot/var/cfengine
+
+
+# CFENGINE_HUB_IP should be provided as an ENV var from the docker run command
+
+
+# entrypoint script for installing and running the agent to be executed from chroot jail
+cat > /chroot/entry.sh << EOF
+#!/bin/bash
+
+set -x
+
+if [ ! -f /var/cfengine/bin/cf-agent ]; then
+  sudo apt install dnsutils
+  cat /etc/resolv.conf
+  dig s3.amazonaws.com
```

Comment:
What's up with the cat and dig?

Reply:
indeed... debugs from past failures no doubt, didn't even see them :)